### PR TITLE
fix: user should not be able to select picture from gallery WPB-90

### PR DESCRIPTION
**Pull request by johnranj**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-90" title="WPB-90" target="_blank">BugWPB-90</a>  [iOS]-Col1: Possible to select picture from library for profile picture
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Users should not be able to select picture from library for profile picture

### Solutions

Removed the option to select photos from library.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift [hunk 1]
```diff
@@ -51,13 +51,6 @@ class ImagePickerManager: NSObject {
                                             message: nil,
                                             preferredStyle: .actionSheet)
 
-        // Choose from gallery
-        let galleryAction = UIAlertAction(title: Alert.choosePicture, style: .default) { [weak self] _ -> Void in
-            self?.sourceType = .photoLibrary
-            self?.getImage(fromSourceType: .photoLibrary)
-        }
-        actionSheet.addAction(galleryAction)
-
```
Comment (agisilaos) [inline wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift]: @johnranj That removes the ability to `select a picture from the gallery` for everyone. I think there's a check that we do if some features are disabled or not and then apply certain changes to the UI. cc: @KaterinaWire

**Review by agisilaos — CHANGES_REQUESTED** [review]

File: wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift]: @johnranj Check `MediaShareRestrictionManager` class

**Review by agisilaos — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift [hunk 2]
```diff
@@ -51,20 +52,22 @@ class ImagePickerManager: NSObject {
...
 
+        guard !MediaShareRestrictionManager(sessionRestriction: ZMUserSession.shared()).isPhotoLibraryEnabled else {
+            let galleryAction = UIAlertAction(title: Alert.choosePicture, style: .default) { [weak self] _ -> Void in
+                self?.sourceType = .photoLibrary
+                self?.getImage(fromSourceType: .photoLibrary)
+            }
+            actionSheet.addAction(galleryAction)
+            return actionSheet
+        }
+
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift]: 1. Any particular reason for changing the order in the sheet?
2. You're missing the cancel button no?

File: wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift [hunk 3]
```diff
@@ -51,20 +52,22 @@ class ImagePickerManager: NSObject {
...
-        actionSheet.addAction(galleryAction)
-
         // Take photo
         let cameraAction = UIAlertAction(title: Alert.takePicture, style: .default) { [weak self] _ -> Void in
             self?.sourceType = .camera
             self?.getImage(fromSourceType: .camera)
         }
         actionSheet.addAction(cameraAction)
 
+        guard !MediaShareRestrictionManager(sessionRestriction: ZMUserSession.shared()).isPhotoLibraryEnabled else {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/Managers/Image/ImagePickerManager.swift]: store the class in ImagePickerManager
```
let mediaShareRestrictionManager = MediaShareRestrictionManager(sessionRestriction: ZMUserSession.shared())
```

**Review by netbe — CHANGES_REQUESTED** [review]

**Review by netbe — APPROVED** [review]
please cherry-pick to release/cycle-3.110 afterwards

**Pull request merged**